### ISSUE TITLE
Better API spec in Swagger UI 

### ISSUE THREAD

**Issue by thiennn** [simplcommerce/SimplCommerce#664]
We have added Swashbuckle to generate specs for SimplCommerce APIs #662. You can now see the generated spec in /swagger. The next step is enhance the APIs spec

**Comment by justindongqiang** [comment]
I’d like to work with this.

**Comment by justindongqiang** [comment]
@thiennn Trying to add the xml comment which generates the xml document displaying in swagger UI, but it does not seem to create the xml doc from xml comments.

**Comment by thiennn** [comment]
Many controllers actions are returning anonymous types, I think we convert to explicit type first to leverage ActionResult<T>.

About the xml doc, we can do it later

**Comment by TATARPRO** [comment]
> 
> 
> Many controllers actions are returning anonymous types, I think we convert to explicit type first to leverage ActionResult.
> 

If I may ask, why are the controllers returning anonymous types instead of strongly typed objects?

**Comment by thiennn** [comment]
Yes, it should return strongly typed. In the early days I return anonymous for faster coding. Now it should be refactored
